atlasexec/atlas.go

```diff
@@ -208,45 +230,47 @@ func (c *Client) Status(ctx context.Context, data *StatusParams) (*StatusReport,
 	if data.RevisionsSchema != "" {
 		args = append(args, "--revisions-schema", data.RevisionsSchema)
 	}
-	var report StatusReport
-	if _, err := c.runCommand(ctx, args, &report); err != nil {
-		return nil, err
-	}
-	return &report, nil
+	return jsonDecode[StatusReport](c.runCommand(ctx, args))
 }
 
 // runCommand runs the given command and unmarshals the output into the given
 // interface.
-func (c *Client) runCommand(ctx context.Context, args []string, report interface{}) (string, error) {
-	cmd := exec.CommandContext(ctx, c.path, args...)
+func (c *Client) runCommand(ctx context.Context, args []string) (io.Reader, error) {
+	var stdout, stderr bytes.Buffer
+	cmd := exec.CommandContext(ctx, c.execPath, args...)
+	cmd.Dir = c.workingDir
 	cmd.Env = append(cmd.Env, "ATLAS_NO_UPDATE_NOTIFIER=1")
-	output, err := cmd.Output()
-	if err != nil {
-		exitErr, ok := err.(*exec.ExitError)
-		if !ok {
-			return string(output), err
-		}
-		if exitErr.Stderr != nil && len(exitErr.Stderr) > 0 {
-			return string(output), &cliError{
-				summary: string(exitErr.Stderr),
-				detail:  string(output),
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
```

Comment:
You can use CombinedOutput

Reply:
Yes, but I want to control the buffer here. So I can return it to the caller.